### DISCUSSION TITLE
Function is not evaluated with x-show directive

### DISCUSSION THREAD

**Discussion by petehouston** [alpinejs/alpine#2263] [7. Bugs (archived)]
After reading the docs, I wonder if `x-show` can evaluate function as the doc doesn't say it.
However, reading the code for `x-show`, I can see it should evaluate if the expression type is a function.

https://github.com/alpinejs/alpine/blob/main/packages/alpinejs/src/directives/x-show.js#L7
https://github.com/alpinejs/alpine/blob/main/packages/alpinejs/src/evaluator.js#L30

I tried with this simple piece of code, and it doesn't work with function. 

```
// alpine 3.4.2 
<script>
    function show() {
      return true;
    }
</script>
<div>
    <p x-show="show()">should SHOW</p>
    <p x-show="!show()">better NOT SHOW </p> <----- this should be hidden but always shown.
</div>
```

I wonder if this is a bug or function is not supported (should not be passed into) for `x-show` directive.

**Comment by SimoTod** [comment]
Your example doesn't have x-data!
